src/MySqlConnector/MySqlClient/MySqlDataReader.cs

```diff
@@ -270,6 +279,100 @@ protected override void Dispose(bool disposing)
+			schemaTable.Locale = CultureInfo.InvariantCulture;
+			schemaTable.MinimumCapacity = colDefinitions.Length;
+
+			var columnName = new DataColumn(SchemaTableColumn.ColumnName, typeof(string));
+			var ordinal = new DataColumn(SchemaTableColumn.ColumnOrdinal, typeof(int));
+			var size = new DataColumn(SchemaTableColumn.ColumnSize, typeof(int));
+			var precision = new DataColumn(SchemaTableColumn.NumericPrecision, typeof(short));
+			var scale = new DataColumn(SchemaTableColumn.NumericScale, typeof(short));
+			var dataType = new DataColumn(SchemaTableColumn.DataType, typeof(System.Type));
+			var providerType = new DataColumn(SchemaTableColumn.ProviderType, typeof(int));
+			var isLong = new DataColumn(SchemaTableColumn.IsLong, typeof(bool));
+			var allowDBNull = new DataColumn(SchemaTableColumn.AllowDBNull, typeof(bool));
+			var isReadOnly = new DataColumn(SchemaTableOptionalColumn.IsReadOnly, typeof(bool));
+			var isRowVersion = new DataColumn(SchemaTableOptionalColumn.IsRowVersion, typeof(bool));
+			var isUnique = new DataColumn(SchemaTableColumn.IsUnique, typeof(bool));
+			var isKey = new DataColumn(SchemaTableColumn.IsKey, typeof(bool));
+			var isAutoIncrement = new DataColumn(SchemaTableOptionalColumn.IsAutoIncrement, typeof(bool));
+			var isHidden = new DataColumn(SchemaTableOptionalColumn.IsHidden, typeof(bool));
+			var baseCatalogName = new DataColumn(SchemaTableOptionalColumn.BaseCatalogName, typeof(string));
+			var baseSchemaName = new DataColumn(SchemaTableColumn.BaseSchemaName, typeof(string));
+			var baseTableName = new DataColumn(SchemaTableColumn.BaseTableName, typeof(string));
+			var baseColumnName = new DataColumn(SchemaTableColumn.BaseColumnName, typeof(string));
+			var isAliased = new DataColumn(SchemaTableColumn.IsAliased, typeof(bool));
+			var isExpression = new DataColumn(SchemaTableColumn.IsExpression, typeof(bool));
+			var isIdentity = new DataColumn("IsIdentity", typeof(bool));
+			ordinal.DefaultValue = 0;
+			isLong.DefaultValue = false;
+
+			// must maintain order for backward compatibility
+			var columns = schemaTable.Columns;
+			columns.Add(columnName);
+			columns.Add(ordinal);
+			columns.Add(size);
+			columns.Add(precision);
+			columns.Add(scale);
+			columns.Add(isUnique);
+			columns.Add(isKey);
+			columns.Add(baseCatalogName);
+			columns.Add(baseColumnName);
+			columns.Add(baseSchemaName);
+			columns.Add(baseTableName);
+			columns.Add(dataType);
+			columns.Add(allowDBNull);
+			columns.Add(providerType);
+			columns.Add(isAliased);
+			columns.Add(isExpression);
+			columns.Add(isIdentity);
+			columns.Add(isAutoIncrement);
+			columns.Add(isRowVersion);
+			columns.Add(isHidden);
+			columns.Add(isLong);
+			columns.Add(isReadOnly);
+
+			for (var i = 0; i < colDefinitions.Length; ++i)
+			{
+				var col = colDefinitions[i];
+				var schemaRow = schemaTable.NewRow();
+				schemaRow[columnName] = col.Name;
+				schemaRow[ordinal] = i;
```

Comment:
This is incompatible with Connector/NET which [uses 1-based ordinals](https://github.com/mysql/mysql-connector-net/blob/5864e6b21a8b32f5154b53d1610278abb3cb1cee/Source/MySql.Data/extensions/NonRT/MySqlDataReader.cs#L1000) for some reason. That seems like a bug to me, and not sure if we should clone that behaviour for backwards compatibility.

Reply:
Let me fix it for backwards compatibility.